source/includes/_feature-extending.md

```diff
@@ -82,4 +84,81 @@ HTTP/1.1 202 Accepted
 
 ```
 
-Personal variables will be ignored and not copied.
\ No newline at end of file
+Personal variables will be ignored and not copied.
+
+### Merging a filtered dataset
+
+If a filter is necessary on the right dataset, it is possible to indicate it
+on the `adapt` expression so that only the selected rows will be brought over
+during the join. This can work in conjunction with selecting a subset of
+variables as well.
+
+To do so, it is necessary to add a `filter` attribute on the payload that can
+contain either a filter expression, wrapped under `{"expression": <expr>}` of
+an existing filter URL wrapped as `{"filter": <url>}` - Note that the filter
+must exist on the right dataset for its expression to match.
+
+Filtering with an expression:
+
+```http
+POST /datasets/{dataset_id}/variables/ HTTP/1.1
+Content-Type: application/json
+
+{
+    "function": "adapt",
+    "args": [{
+        "dataset": "/datasets/{other_id}/"
+    }, {
+        "variable": "/datasets/{other_id}/variables/{right_key_id}/"
+    }, {
+        "variable": "/datasets/{dataset_id}/variables/{left_key_id}/"
+    }]
+    "filter": {
+        "expression": {
+            "function": "==",
+            "args": [
+                {"variable": {right_dataset_variable}},
+                {"value": <value>}
+            ]
+        }
+    }
+}
+```
+
+Filtering with a filter URL and selecting variables:
+
+```http
+POST /datasets/{dataset_id}/variables/ HTTP/1.1
+Content-Type: application/json
+
+{
+    "function": "select",
+    "args": [{
+        "map": {
+            "{right_var1_id}/": {
+                "variable": "/datasets/{other_id}/variables/{right_var1_id}/"
+            },
+            "{right_var2_id}/": {
+                "variable": "/datasets/{other_id}/variables/{right_var2_id}/"
+            },
+            "{right_var3_id}/": {
+                "variable": "/datasets/{other_id}/variables/{right_var3_id}/"
+            }
+        }
+    }],
+    "frame": {
+        "function": "adapt",
+        "args": [{
+            "dataset": "/datasets/{other_id}/"
+        }, {
+            "variable": "/datasets/{other_id}/variables/{right_key_id}/"
+        }, {
+            "variable": "/datasets/{dataset_id}/variables/{left_key_id}/"
+        }]
+    },
```

Comment:
I am putting the filter on the top level of the expression always. In reality during munging I am parsing this filter parameter and injecting it deeper inside the adapt zcl where it corresponds (in two places): https://github.com/Crunch-io/backend-http/pull/372/files#diff-22a7d09e5fcd405b017fe92313240ff6R647

I understand it as a filter on whatever you are adapting, be it a single adapt function or a more complex nesting of funcs.